### PR TITLE
Move test to pytest

### PR DESCRIPTION
**Pull request by davzucky**
This is a refactoring of all the test to use pytest and fixture. Just wanted to clean the test before implementing the custom format

### CODE REVIEW

File: tests/test_prettytable.py [hunk 1]
```diff
@@ -125,293 +151,424 @@ def testDeleteColumn(self):
...
+@pytest.fixture(scope="function")
+def city_data_from_csv():
+    csv_string = """City name, Area , Population , Annual Rainfall
+    Sydney, 2058 ,  4336374   ,      1214.8
+    Melbourne, 1566 ,  3806092   ,       646.9
+    Brisbane, 5905 ,  1857594   ,      1146.4
+    Perth, 5386 ,  1554769   ,       869.4
+    Adelaide, 1295 ,  1158259   ,       600.5
+    Hobart, 1357 ,   205556   ,       619.5
+    Darwin, 0112 ,   120900   ,      1714.7"""
```
Comment (hugovk) [inline tests/test_prettytable.py]: Please could you remove the spaces before commas, and put just one after commas?

File: tests/test_prettytable.py [hunk 2]
```diff
@@ -547,10 +686,20 @@ def testAsciiBreakLine(self):
...
+        self, rows: List[List[Any]], hrule: int, expected_result: str
+    ):
+        t = PrettyTable(["Field 1", "Field 2"])
+        for row in rows:
+            t.add_row(row)
+        result = t.get_string(hrules=hrule)
+        assert result.strip() == expected_result.strip()
 
-    def testHtmlBreakLine(self):
+    def test_break_line_HTLM(self):
```
Comment (hugovk) [inline tests/test_prettytable.py]:
```suggestion
    def test_break_line_HTML(self):
```

File: tests/test_prettytable.py [hunk 3]
```diff
@@ -125,293 +151,424 @@ def testDeleteColumn(self):
...
+#         )
+#         self.cur.execute('INSERT INTO cities VALUES ("Darwin", 112, 120900, 1714.7)')
+#         self.cur.execute('INSERT INTO cities VALUES ("Hobart", 1357, 205556, 619.5)')
+#         self.cur.execute('INSERT INTO cities VALUES ("Sydney", 2058, 4336374, 1214.8)')
+#         self.cur.execute(
+#             'INSERT INTO cities VALUES ("Melbourne", 1566, 3806092, 646.9)'
+#         )
+#         self.cur.execute('INSERT INTO cities VALUES ("Perth", 5386, 1554769, 869.4)')
+#         self.cur.execute("SELECT * FROM cities")
+#         self.x = from_db_cursor(self.cur)
```
Comment (hugovk) [inline tests/test_prettytable.py]: Remove this old commented code?

**Review by hugovk — CHANGES_REQUESTED** [review]
Thank you for this! A few comments.

File: tests/test_prettytable.py (hunk 3 above)
Comment (davzucky) [reply tests/test_prettytable.py]: Done

File: tests/test_prettytable.py (hunk 2 above)
Comment (davzucky) [reply tests/test_prettytable.py]: Done

File: tests/test_prettytable.py (hunk 1 above)
Comment (davzucky) [reply tests/test_prettytable.py]: Done

**Review by hugovk — CHANGES_REQUESTED** [review]
Please also fix the linting:

https://github.com/jazzband/prettytable/pull/135/checks?check_run_id=3826690535

The isort can be done automatically:

`python -m pip install -U pre-commit && pre-commit run --all-files`

(And optionally `pre-commit install` to have it check your changes when you commit.)

The Flake8 things need to be done manually, I guess these are mostly things to do like `noqa: X123`.

(Another idea: it could be worth splitting the test file into smaller ones, but you don't need to do this, and probably something for a followup PR anyway.)

**Comment by davzucky** [comment]
Thank you for all these information. I will add that on the Doc as well later.
Fixed the linter errors.

Yes it can be split. however the question could be how ? What could be a good per use case split.

**Review by hugovk — APPROVED** [review]

**Comment by hugovk** [comment]
> Yes it can be split. however the question could be how ? What could be a good per use case split.

Now that's a good question... Perhaps some related test classes in the same file, and the rest stay where they are? If there's no obvious way, I wouldn't worry about it too much.

Thanks again!

**Pull request merged**